runtime/compiler/control/CompilationThread.cpp

```diff
@@ -6269,9 +6352,59 @@ TR::CompilationInfoPerThreadBase::preCompilationTasks(J9VMThread * vmThread,
             *aotCachedMethod = javaVM->sharedClassConfig->findCompiledMethodEx1(vmThread, J9_ROM_METHOD_FROM_RAM_METHOD(method), &flags);
             if (!(flags & J9SHR_AOT_METHOD_FLAG_INVALIDATED))
                {
+#ifdef COMPRESS_AOT_DATA
+               TR_AOTMethodHeader *aotMethodHeader = (TR_AOTMethodHeader*)(((J9JITDataCacheHeader*)*aotCachedMethod) + 1);
+               if (aotMethodHeader->flags & TR_AOTMethodHeader_CompressedMethodInCache)
+                  {
+                  int dataSize = *((uint32_t*)(*aotCachedMethod)-2);
+                  int decompressedDataSize = aotMethodHeader->compileMethodDataSize + aotMethodHeader->compileMethodCodeSize;
+                  void *decompressedData = trMemory.allocateHeapMemory(decompressedDataSize);
+                  int uncompressedSize = sizeof(J9JITDataCacheHeader) + sizeof(TR_AOTMethodHeader);
```

Comment:
Nitpick, but there should be consistency regarding "decompressed" and "uncompressed"

Reply:
Same comment as below, rename this variable to refer to the fact that this is the size of the header